### DISCUSSION TITLE
Openvpn site to site peer slow

### DISCUSSION THREAD

**Discussion by gvolponi** [Ysurac/openmptcprouter#2847] [General]
installed 2 OMR with 2 VPS on 2 different site, seems working perfectly with standard traffic with good aggregation and bonding. I have setup on each site pfsense and configured openvpn for create a site to site vpn. 
The VPN with openvpn is slow, seems use only 1 connection. It's normal? there are some extra configuration for optimize traffic. The openvpn use UDP.
Thanks

**Comment by iambenmitchell** [comment]
> The VPN with openvpn is slow, seems use only 1 connection. It's normal? there are some extra configuration for optimize traffic. The openvpn use UDP.


I've had nothing but issues trying to setup additional VPNs on OMR. 

I would just buy two ER605 from TP-Link or something similar and do it from the LAN

**Reply by gvolponi** [reply]
Tried with IPsec also and the performance are very poor...
On one site there are 4 lines with total speed of 550 down and 90 up, on the other site 3 lines with a total of 180 down and 40 up.
The VPN with iperf test say max 15Mbits.
Tried with only 1 line on the second site, without OMR and speed are 18Mbits....
Very very strange...

**Comment by Ysurac** [comment]
You should use OpenVPN in TCP mode. By default TCP traffic use the proxy, that is faster, and UDP use a VPN that is lower (and in OpenVPN case make a VPN over a VPN).

**Reply by gvolponi** [reply]
Thanks for the advice, I'll give it a try and update you.

**Reply by gvolponi** [reply]
Tried to switch openvpn from udp to TCP but no lucky.  You can see in thi image [http://is.am/5f4u](http://is.am/5f4u) that only WAN1 interface is used. What can I try to improve performance?

**Comment by gvolponi** [comment]
![OMR_TEST](https://github.com/Ysurac/openmptcprouter/assets/24852459/9813ad6c-8bc1-41a7-a5ed-2e4bd8ada826)
Tried change congestion control from cubic to bbr or bbr2 no lucky
As you can see in the image attached, if I do a speed test the upload function very weel, with iperf through OpenVPN only the principal wan interface seems to work
